src/machines/java/com/enderio/machines/common/io/fluid/MachineFluidTank.java

```diff
@@ -0,0 +1,250 @@
+package com.enderio.machines.common.io.fluid;
+
+import com.enderio.EnderIO;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BucketItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraftforge.common.capabilities.ForgeCapabilities;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidType;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import net.minecraftforge.fluids.capability.templates.FluidTank;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class MachineFluidTank extends FluidTank {
+
+    public static final int DEFAULT_CAPACITY = 8 * FluidType.BUCKET_VOLUME;
+    public boolean allowInput = true, allowOutput = true;
+    private @Nullable BlockEntity parent = null;
+
+    public MachineFluidTank(int capacity, BlockEntity parent) {
+        super(capacity, (FluidStack stack)->true);
+        this.parent = parent;
+    }
+    public MachineFluidTank(BlockEntity parent) {
+        super(DEFAULT_CAPACITY, (FluidStack stack)->true);
+        this.parent = parent;
+    }
+
+    /**
+     * @param capacity - the capacity of the tank
```

Comment:
When writing JavaDocs, you don't need this hyphen as that'll then be rendered in with the parameter descriptions in an IDE.